### ISSUE TITLE
FactoryVec[Deque] update logic is limiting

### ISSUE THREAD

**Issue by uvizhe** [Relm4/Relm4#163]
Imagine this factory collection:
```
0: Foo { leader = true }  (Object A)
1: Foo { leader = false } (Object B)
```
What if I want to implement logic where I need to remove `leader` from a list and set new `leader`?
This means removing element at index 0 and updating element at index 1.
Desired result collection would look like this
```
0: Foo { leader = true }  (Object B)`
```
However, due to how Factory collection update logic works it's impossible to do so.
Updating an element after removal at the same index is [prohibited](https://github.com/Relm4/Relm4/blob/main/src/factory/collections/factory_vec_deque.rs#L170).
Updating an element at index 1 while removing element at index 0 is useless because it won't invoke update at index 0.

One possible fix I see is to call `view()` in `generate()` function for every `index` where there was removal and that is still present in a collection`.

**Comment by AaronErhardt** [comment]
That's definitely an error in the update logic. This behavior shouldn't happen.

One reason why I changed the implementation of `FactoryVecDeque` for the upcoming version was exactly because the update logic in 0.4 is very hard to understand and also flawed as this issue shows. Yet, I don't know if it's worth to fix this now that 0.5 should soon see its first beta release.

I just have one question:
>  This means removing element at index 0 and updating element at index 1.

Did you mean this: This means removing element at index 0 and updating element at index 0 ?
Because one the first element is removed the second one is now the first and has index 0.

**Comment by AaronErhardt** [comment]
0.5 is now stable which should fix this issue.
